packages/wallet/test/SingleAddressWallet/load.test.ts

```diff
@@ -390,3 +391,31 @@ describe('SingleAddressWallet creates big UTXO', () => {
     wallet.shutdown();
   });
 });
+
+describe('SingleAddressWallet.error$', () => {
+  it('emits non retryable errors', async () => {
+    const stores = createInMemoryWalletStores();
+    const utxoSet = generateUtxos(30, 10);
+
+    const wallet = await createWallet(stores, {
+      chainHistoryProvider: mocks.mockChainHistoryProvider(),
+      networkInfoProvider: {
+        ...mocks.mockNetworkInfoProvider(),
+        ledgerTip: jest.fn().mockRejectedValue(new InvalidStringError('Test invalid string error'))
+      },
+      rewardsProvider: mocks.mockRewardsProvider(),
+      utxoProvider: mocks.mockUtxoProvider(utxoSet)
+    });
+
+    // wallet.error$ must be observed till the beginning of time
+    const errorPromise = expect(firstValueFrom(wallet.error$)).resolves.toBeInstanceOf(InvalidStringError);
+
+    wallet.tip$.subscribe(() => {
+      throw new Error('Expected tip$ to not emit');
+    });
+
+    await errorPromise;
+
+    wallet.shutdown();
```

Comment:
A more expressive way to assert it doesn't emit:
```suggestion
    let tipHandler = jest.fn();
    wallet.tip$.subscribe(tipHandler);
    await errorPromise;
    wallet.shutdown();
    expect(tipHandler).not.toBeCalled();
```

Reply:
efd3bd734 should solve this